pkg/policy/k8s/service_test.go

```diff
@@ -732,3 +735,40 @@ func Test_hasMatchingToServices(t *testing.T) {
 		})
 	}
 }
+
+func Test_serviceNotificationsQueue(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	upstream := make(chan k8s.ServiceNotification)
+	downstream := serviceNotificationsQueue(ctx, stream.FromChannel(upstream))
+
+	// Test that sending events in upstream does not block on unbuffered channel
+	upstream <- k8s.ServiceNotification{ID: k8s.ServiceID{Name: "svc1"}}
```

Comment:
Thanks for the suggestion! I wanted to make sure we do an e2e test, rather than testing enqueue/dequeue. The important behavior to assert is that it doesn't block the producer, thereby solving the deadlock